sample-apps/celery_sample/app/tasks.py

```diff
@@ -0,0 +1,17 @@
+import random
+import time
+
+from app.settings import BaseConfig
+from celery import Celery
+
+from judoscale.celery import judoscale_celery
+
+celery = Celery("CelerySampleApp", broker="redis://localhost:6379/0")
+
+judoscale_celery(celery, extra_config=BaseConfig.JUDOSCALE)
```

Comment:
I'm not a huge fan of this, if I'm being honest.

The end user should be able to override the Celery integration configuration, but I didn't want to make the mechanism specific on any particular web framework or anything like that. Even though in 99% of the cases it _will_ co-exist with a web framework I don't think we should be making assumptions about how the user will use this integration.

It'd be nice if the Celery integration could configure itself from the same configuration as the web app (`app.config` for Flask and `django.conf.settings` for Django), but I've left it like this for now.

Reply:
I'm sort of surprised there's not already some kind of convention for this in Python.

In Ruby, for example, it's common for a library to have a global configuration mechanism that can be shared amongst multiple process. For `judoscale-ruby`, you [configure it](https://github.com/judoscale/judoscale-ruby#configuration) like this:

```ruby
Judoscale.configure do |config|
  config.report_interval_seconds = 5
end
```

As long as that code is executed by by the web and worker processes, they'll share the configuration. In a Rails app, that usually means you put it in an "initializer" (`/config/initializers/judoscale.rb`), and it will be auto-loaded by Rails.

But if there's no convention like that in Python, I think the way you have it here is fine.

Reply:
Indeed — the Ruby approach is very nice. I'm not aware of any convention in Python that does this. One other option is to provide framework-specific ways to initialise this integration, e.g an extension for Flask, an app for Django, etc.

Reply:
On second thought, it looks like the way Celery recommends initialising the broker for Django is [fairly simplistic](https://docs.celeryq.dev/en/stable/django/first-steps-with-django.html), so I'm not even sure adding a "Judoscale Django Celery" app would work. [Same for Flask](https://flask.palletsprojects.com/en/2.2.x/patterns/celery/).

Reply:
No problem. I'm okay with this config setup.